src/Tizen.Applications.WatchfaceComplication/Tizen.Applications/DesignElement.cs

```diff
@@ -0,0 +1,219 @@
+﻿/*
+ * Copyright (c) 2018 Samsung Electronics Co., Ltd All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Tizen.Applications.WatchfaceComplication
+{
+    /// <summary>
+    /// Represents the DesignElement class for the watch application.
+    /// </summary>
+    /// <since_tizen> 6 </since_tizen>
+    public abstract class DesignElement : IEditable
+    {
+
+        private IEnumerable<Bundle> _candidates;
+        private int _currentDataIndex;
+        private Highlight _highlight;
+        private string _editableName;
+        private int _editableId;
+
+        /// <summary>
+        /// Initializes the DesignElement class.
+        /// </summary>
+        /// <param name="candidates">The candidates list.</param>
+        /// <param name="currentDataIndex">The currently selected data index of candidate list.</param>
```

Comment:
I don't know exactly what currentDataIndex is with this explain.

Suggestion:
The index of selected data among candidate list.

If the parameter means my suggestion, please change it.